### ISSUE TITLE
Spotify broken with hosts file

### ISSUE THREAD

**Issue by olivercamos** [StevenBlack/hosts#552]
Using Unified Hosts for magisk module with standard host files, no additions.

With the hosts enabled, I cannot access the Spotify home screen, radio or search pages.

Disabling all hosts in magisk instantly cures the issue.

I tried to add a whitelist wildcard of the word 'spotify' with no success.

**Comment by VincentJoshuaET** [comment]
Also experienced same issue. I downloaded the latest hosts file (unified with no other add-ons) and pushed it to /system/etc/ using TWRP.

The domain "spclient.wg.spotify.com" seems to exist in two lines. One is commented out (with "#"), but one is not. I suspect that's the reason.

**Comment by VincentJoshuaET** [comment]
Update: I commented out the line I mentioned and Spotify is now working.

**Comment by olivercamos** [comment]
Worked a treat, thanks for that. Hopefully the host sourcr will be fixed in due course.

**Comment by FadeMind** [comment]
Domain **spclient.wg.spotify.com** cannot to be deleted cause without blocking this domain ads appear in Spotify WebApp and Spotify client (Linux).

Spotify and others websites (mostly streaming) are fully functional on my side with current state of hosts AND this whitelist file:
```
2mdn.net
adocean.pl
criteo.com
facebook.net
fbcdn.net
flarecloud.net
gemius.pl
googleapis.com
iplsc.com
npttech.com
onet.hit.gemius.pl
openload.co
openload.io
pro.hit.gemius.pl
s.tvp.pl
t.scdn.co
```

**Comment by VincentJoshuaET** [comment]
@FadeMind do you have any suggestions for Android users? [Screenshot](https://i.imgur.com/snX8Nq6.png) on the original unedited hosts file, it's mentioned that the Spotify domain affects the mobile app.

_Sent from my ONEPLUS A3003 using [FastHub](https://play.google.com/store/apps/details?id=com.fastaccess.github)_

**Comment by StevenBlack** [comment]
Working fine here on MacOS.

Anyone have a theory about why this would be critical on 'Droid but not on *nix?

**Comment by VincentJoshuaET** [comment]
Not sure, but it seems that this issue has always been coming back and forth: https://github.com/StevenBlack/hosts/issues/270

**Comment by StevenBlack** [comment]
Any suggestions Vincent?  @VincentJoshuaET 

Somebody please trace responses from `spclient.wg.spotify.com`.  I'm thinking of wiring a mock.

**Comment by VincentJoshuaET** [comment]
Maybe a different hosts file for Android/iOS?

**Comment by FadeMind** [comment]
> Maybe a different hosts file for Android/iOS?
LGTM @StevenBlack

**Comment by StevenBlack** [comment]
iOS doesn't have fungible hosts, so this would be for Android only.

**Comment by rprimus** [comment]
Thu Apr  5 18:49:34 BST 2018

I've just run into this issue (Spotify home page not loading) and found the two lines in `/system/etc/hosts`.

### Suggestion:

As most (Android) users use the [`Magisk`](https://github.com/Zackptg5/Unified-Hosts-Adblock)  module, this can be solved there by requesting the module author (@Zackptg5) to:
1.  Add a default whitelist containing `spclient.wg.spotify.com` (or append it if a whitelist already exists), or
2.  As `/system/bin/hosts` is a bash script, modify it to run `sed -i '/spclient.wg.spotify.com/d /system/etc/hosts` -  after `/system/etc/hosts` is updated.

**Comment by notDavid** [comment]
@vaguerant  _"Breaking web services on some platforms to benefit other platforms is going to leave a substantial proportion of users dissatisfied."_

I think it's not just some platforms vs other platforms, it's also "paying Spotify users" vs "non paying Spotify users (who are being served ads)".

I personally believe that **"not breaking apps"** should be the higher priority.
By not fixing this, we are basically breaking things for some (paying) Spotify users only to achieve that non-paying Spotify users can circumvent ads.
So blocking ads now has a higher priority than "not breaking things".

That can't be right... 🤷🏼‍♂️ Or perhaps i'm missing something 😊

Personally, i would vote for creating a separate list for this, because then people are consciously blocking it and will easier realise why it's not working.

**Comment by StevenBlack** [comment]
This is fixed in [Release 1.3.17](https://github.com/StevenBlack/hosts/releases).

Thanks everybody.
